src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -857,10 +864,70 @@ def update_gui(self):
             self.txtZstepsize.setText(GuiUtils.formatValue(self.mag_sld_data.zstepsize))
         # otherwise leave as set since editable by user
 
+        # update the value of the Radius of Gyration with values from the loaded data
+        if self.is_nuc:
+            if self.nuc_sld_data.is_elements:
+                self.txtROG.setText(str("N/A for Elements"))
+            else:
+                self.txtROG.setText(str(round(self.radius_of_gyration(),1)) + " Å")
+        elif self.is_mag:
+            self.txtROG.setText(str("N/A for magnetic data"))
+        else:
+            self.txtROG.setText(str("N/A for no data"))
+            
+
         # If nodes or stepsize changed then this may effect what values are allowed
         self.gui_text_changed(sender=self.txtNoQBins)
         self.gui_text_changed(sender=self.txtQxMax)
     
+    def radius_of_gyration(self):
+        #Calculate Center of Mass(CoM) First
+        CoM = self.centerOfMass()
+
+        #Now Calculate RoG
+        RoGNumerator = RoGDenominator = 0.0
+
+        for i in range(len(self.nuc_sld_data.pos_x)):
+            coordinates = [0.0,0.0,0.0]
+            coordinates = [float(self.nuc_sld_data.pos_x[i]),float(self.nuc_sld_data.pos_y[i]),float(self.nuc_sld_data.pos_z[i])]
+            
+            #Coh b - Coherent Scattering Length(fm)
+            cohB = periodictable.elements.symbol(self.nuc_sld_data.pix_symbol[i]).neutron.b_c
+
+            #Calculate the Magnitude of the Coordinate vector for the atom and the center of mass
+            MagnitudeOfCoM = numpy.sqrt(numpy.power(CoM[0]-coordinates[0],2) + numpy.power(CoM[1]-coordinates[1],2) + numpy.power(CoM[2]-coordinates[2],2))
+
+            #Calculate Rate of Gyration (Squared) with the formular
+            RoGNumerator += cohB * (numpy.power(MagnitudeOfCoM,2))
+            RoGDenominator += cohB
+
+        RoG = numpy.sqrt(RoGNumerator/RoGDenominator)
+
+        return (RoG)
+    
+    def centerOfMass(self):
+        """Calculate Center of Mass(CoM) of provided atom"""
+        CoMnumerator= [0.0,0.0,0.0]
+        CoMdenominator = [0.0,0.0,0.0]
+        CoM = [0.0,0.0,0.0]
+
+        for i in range(len(self.nuc_sld_data.pos_x)):
+            coordinates = [0.0,0.0,0.0]
+            coordinates = [float(self.nuc_sld_data.pos_x[i]),float(self.nuc_sld_data.pos_y[i]),float(self.nuc_sld_data.pos_z[i])]
+            
+            #Coh b - Coherent Scattering Length(fm)
+            cohB = periodictable.elements.symbol(self.nuc_sld_data.pix_symbol[i]).neutron.b_c
+
+            for i in range(3): #sets CiN
+                CoMnumerator[i] += (coordinates[i]*cohB)
+                CoMdenominator[i] += cohB
```

Comment:
A more "pythonic" way to fill arrays sequentially is with `.append`.
```
            for i in range(3): #sets CiN
                CoMnumerator.append(coordinates[i]*cohB)
                CoMdenominator.append(cohB)
```
This also means you can simplify definitions of CoMnumerator etc. to empty lists,
```
        CoMnumerator= []
        CoMdenominator = []
        CoM = []
```